### ISSUE TITLE
How to include client specific claims in claim

### ISSUE THREAD

**Issue by ebinroy** [damienbod/angular-auth-oidc-client#1029]
Hi All,
      I am using Identity Server 4 as STS and would like to implement claim feature

User claims are stored in STS database but user claims specific to the clients are defined in ClientDB (eg: role) which can get through API

In .Net core this can be achieved using the below example

startup,.cs --> [code in startup class](https://github.com/KevinDockx/DealingWithCredentialsWhenSecuringAspNetCore3/blob/485171e400553955f97935c69d8fe7bea5ab42bc/Finished%20sample/src/ImageGallery.Client/Startup.cs#L119)

Implementation of adding client user claims --> [add client specific claims](https://github.com/KevinDockx/DealingWithCredentialsWhenSecuringAspNetCore3/blob/485171e400553955f97935c69d8fe7bea5ab42bc/Finished%20sample/src/ImageGallery.Client/PostConfigurationOptions/OpenIdConnectOptionsPostConfigureOptions.cs#L16)

Basically I want to add client claims with the claims

How this can be achieved using this library.

**Comment by damienbod** [comment]
I would do this in the IProfileService on your STS. But if you want to add 'claims' to the angular app, then you want need to do to an API call using the access token and get the extra claims.

Greetings Damien

**Comment by ebinroy** [comment]
Thanks for your reply.

As per my understanding, I already added user claims using IProfileService but there are certain claims that is specific for a client. That should  be added on client side ( if angular app then shd be done on angular side) So the user first authenticate need to include those claims with  oidcSecurityService.userData$

is there any option to add those claims with userData$
